added importer for narrowband pattern along with related demo

pawarsp opened the pull request.

pawarsp committed:
added importer for narrowband pattern along with related demo

SebastianSemper commented on demo/patternNBandImport.py:
```diff
@@ -0,0 +1,76 @@
+if __name__ == "__main__":
+
+    # directory location and name of pattern .mat file
+    patternDir = "TUI_SPUCA_2x8_I_20MHz_pattern.mat"
```
variable name is kinda misleading, since it is not a directory, but a file

pawarsp replied:
if the file isn't in the same folder as the demo file, we would need to add the whole directory path, right? That isn't the case - so we can replace it with patternFile?

MichaelDoebereiner replied:
you could name it "patternPath". I think this should be clear then

SebastianSemper commented on demo/patternNBandImport.py:
```diff
@@ -0,0 +1,76 @@
+
+    # desired EADF object
+    _, EADF = eadf.importers.fromIrregularAngDataNBandMatFile(
+        patternDir, **kwargs
```
you can just write arrPorIndUser=arrPorInd, arrPolUser=arrPol instead of creating a dictionary first.

pawarsp replied:
alright!

SebastianSemper commented on demo/patternNBandImport.py:
```diff
@@ -0,0 +1,76 @@
+    )
+
+    numAzi = 60
+    numEle = 30
```
I always try to use the name CoEle instead of just Ele to avoid any confusion.

pawarsp replied:
good point.

SebastianSemper commented on demo/patternWBandImport.py:
```diff
@@ -0,0 +1,81 @@
+
+    # desired EADF object
+    _, EADF = eadf.importers.fromIrregularAngDataWBandMatFile(
+        patternDir, **kwargs
```
See the other file's comments.

SebastianSemper commented on eadf/__init__.py:
```diff
@@ -160,7 +160,10 @@
 from .importers import fromAngleListData, fromFourierData, fromArraydefData
 
-from .importers import fromIrregularAngDataMatFile
+from .importers import (
```
Nice! Did not know that this works. :3

SebastianSemper commented on eadf/__init__.py:
```diff
@@ -160,7 +160,10 @@
 
-from .importers import fromIrregularAngDataMatFile
+from .importers import (
+    fromIrregularAngDataNBandMatFile,
```
What is so irregular about the angular data anyway? I don't understand the function name, but I also cannot offer anything better.

pawarsp replied:
By irregular, it meant not in the regular format expected for the EADF.

SebastianSemper replied:
Ah, I see. In that case this is not necessary, bc if it were regular in that sense we would not write an importer anyway.

pawarsp replied:
Updated to fromWidebandAngData and fromNarrowbandAngData.

SebastianSemper commented on eadf/importers.py:
```diff
@@ -198,7 +202,7 @@ def fromAngleListData(
 
 
-def fromIrregularAngDataMatFile(path: str, **kwargs) -> EADF:
+def fromIrregularAngDataWBandMatFile(path: str, **kwargs) -> EADF:
```
WBand could be changed to Wideband, but again, I don't get the function name...

SebastianSemper reviewed (changes requested):
Only minor stuff, we are basically ready to merge. :3

pawarsp committed:
revised importers for importing pattern data

SebastianSemper commented on demo/patternWBandImport.py:
```diff
@@ -17,7 +17,7 @@
 
 >>> cd demo
->>> python patternImport.py
+>>> python patternWBandImport.py
```
are you sure that this was your only intended commit?

pawarsp committed:
updated importers.py and added demo patternNBandImport.py

pawarsp committed:
updated file and variable names for pattern file importers and respective demos

MichaelDoebereiner reviewed (approved):
as Sebastian has already mentioned some changes, you get the go from my side ;)

SebastianSemper committed:
Extend Importers

1. Added input checks for all file-based importers
2. changed the interface of some importers to return
   a normal eadf object and not tuple, adjusted
   the demos accordingly.
